funcx_endpoint/funcx_endpoint/endpoint/endpoint_manager.py

```diff
@@ -0,0 +1,365 @@
+        with open(endpoint_config_target_file, "w") as w:
+            w.write(endpoint_config_template)
+
+        return endpoint_dir
+
+    def configure_endpoint(self, name, endpoint_config):
+        self.name = name
+        endpoint_dir = os.path.join(self.funcx_dir, self.name)
+        new_config_file = os.path.join(endpoint_dir, self.funcx_config_file_name)
+
+        if not os.path.exists(endpoint_dir):
+            self.init_endpoint_dir(endpoint_config=endpoint_config)
+            print(f'A default profile has been create for <{self.name}> at {new_config_file}')
+            print('Configure this file and try restarting with:')
+            print(f'    $ funcx-endpoint start {self.name}')
+        else:
+            print(f'config dir <{self.name}> already exsits')
+            raise Exception('ConfigExists')
+
+    def init_endpoint(self):
+        """Setup funcx dirs and default endpoint config files
+
+        TODO : Every mechanism that will update the config file, must be using a
+        locking mechanism, ideally something like fcntl https://docs.python.org/3/library/fcntl.html
+        to ensure that multiple endpoint invocations do not mangle the funcx config files
+        or the lockfile module.
+        """
+        _ = FuncXClient()
+
+        if os.path.exists(self.funcx_config_file):
+            typer.confirm(
+                "Are you sure you want to initialize this directory? "
+                f"This will erase everything in {self.funcx_dir}", abort=True
+            )
+            self.logger.info("Wiping all current configs in {}".format(self.funcx_dir))
+            backup_dir = self.funcx_dir + ".bak"
+            try:
+                self.logger.debug(f"Removing old backups in {backup_dir}")
+                shutil.rmtree(backup_dir)
+            except OSError:
+                pass
+            os.renames(self.funcx_dir, backup_dir)
+
+        if os.path.exists(self.funcx_config_file):
+            self.logger.debug("Config file exists at {}".format(self.funcx_config_file))
+            return
+
+        try:
+            os.makedirs(self.funcx_dir, exist_ok=True)
+        except Exception as e:
+            print("[FuncX] Caught exception during registration {}".format(e))
+
+        shutil.copyfile(self.funcx_default_config_template, self.funcx_config_file)
+        self.init_endpoint_dir()
+
+    def check_endpoint_json(self, endpoint_json, endpoint_uuid):
+        if os.path.exists(endpoint_json):
+            with open(endpoint_json, 'r') as fp:
+                self.logger.debug("Connection info loaded from prior registration record")
+                reg_info = json.load(fp)
+                endpoint_uuid = reg_info['endpoint_id']
+        elif not endpoint_uuid:
+            endpoint_uuid = str(uuid.uuid4())
+        return endpoint_uuid
+
+    def start_endpoint(self, name, endpoint_uuid, endpoint_config):
+        self.name = name
+        self.endpoint_config = endpoint_config
```

Comment:
Do we need to save these as instance vars? Are they used in other places outside of start?
At a minimum, these should be moved to explicit setter methods. It's surprising to see this method change the instance.

Reply:
Removed the member variable. The variable is needed for `daemon_launch`. So after the change, the variable is passed to `daemon_launch` as parameter.